edgelet/edgelet-http-mgmt/src/client/module.rs

```diff
@@ -29,9 +30,9 @@ impl ModuleClient {
     pub fn new(url: &Url) -> Result<ModuleClient, Error> {
         let client = Client::builder().build(UrlConnector::new(url)?);
 
-        let base_path = get_base_path(url);
+        let base_path = get_base_path(url)?;
         let mut configuration = Configuration::new(client);
-        configuration.base_path = base_path.to_string();
+        configuration.base_path = base_path.to_str().unwrap().to_string();
```

Comment:
It's more likely to not be a valid `str` on Windows compared to Linux. Can you change `configuration.base_path` to an `OsString` ?

Reply:
As discussed, this gets parsed as a `url::Url`, so it does need to stay as a `String` rather than `OsString` / `PathBuf` and needs to be valid UTF-8.

So change the `unwrap()` to an `expect("Path should be valid utf-8")` or something.

Reply:
Will do.

Reply:
Update: if the original URL is not valid UTF-8, it will get encoded by `Url::parse` and we won't know there's a problem until we call `get_base_path(url).to_str()`. So `expect` isn't enough; we should return an error.